dotnet/src/Connectors/Connectors.MistralAI.UnitTests/Client/MistralClientTests.cs

```diff
@@ -238,117 +246,236 @@ await foreach (var chunk in response)
+        kernel.Plugins.AddFromType<WeatherPlugin>();
+
+        var invokedFunctions = new List<string>();
+        var filter = new FakeAutoFunctionFilter(async (context, next) =>
         {
-            Assert.Equal(this.RequestUri, request.RequestUri);
-            Assert.Equal(this.Method, request.Method);
-            Assert.Equal(this.RequestHeaders, request.Headers);
-
-            this.RequestContent = await request.Content!.ReadAsStringAsync(cancellationToken);
-
-            if (this._responseString is not null)
-            {
-                this.ResponseMessage = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
-                {
-                    Content = new StringContent(this._responseString, System.Text.Encoding.UTF8, "application/json")
-                };
-            }
-            if (this._responseBytes is not null)
-            {
-                this.ResponseMessage = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
-                {
-                    Content = new StreamContent(new MemoryStream(this._responseBytes))
-                };
-            }
-
-            return await Task.FromResult(this.ResponseMessage);
-        }
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+            context.Terminate = true;
+        });
+        kernel.AutoFunctionInvocationFilters.Add(filter);
+
+        // Act
+        var executionSettings = new MistralAIPromptExecutionSettings { ToolCallBehavior = MistralAIToolCallBehavior.AutoInvokeKernelFunctions };
+        var chatHistory = new ChatHistory
+        {
+            new ChatMessageContent(AuthorRole.User, "What is the weather like in Paris?")
+        };
+        var response = await client.GetChatMessageContentsAsync(chatHistory, default, executionSettings, kernel);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.Single(response);
+        Assert.Equal("12°C\nWind: 11 KMPH\nHumidity: 48%\nMostly cloudy", response[0].Content);
+        Assert.Null(response[0].ModelId);
+        Assert.Equal(1, this.DelegatingHandler.SendAsyncCallCount);
+        Assert.Equal(3, chatHistory.Count);
+        Assert.Contains("GetWeather", invokedFunctions);
     }
 
+    #region internal classes
     public sealed class WeatherPlugin
     {
         [KernelFunction]
         [Description("Get the current weather in a given location.")]
         public string GetWeather(
-            [Description("The city and department, e.g. Marseille, 13")] string location,
-            [Description("The temperature units one of celsius or fahrenheit")] TemperatureUnit units
-            ) => $"{{\"location\": \"{location}\", \"unit\": \"{units}\"}}";
+            [Description("The city and department, e.g. Marseille, 13")] string location
+            ) => "12°C\nWind: 11 KMPH\nHumidity: 48%\nMostly cloudy";
+    }
+
+    internal enum TemperatureUnit { Celsius, Fahrenheit }
+
+    public class WidgetFactory
```

Comment:
```suggestion
    private sealed class WidgetFactory
```

Reply:
Causes potentially uninstatiated class erros

Reply:
If the test works with them in private, pragma ignore them, 
The analyzer can't predict some class usages like serialization.